presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/TableLayoutRewriter.java

```diff
@@ -115,4 +123,14 @@ public PlanNode planTableScan(TableScanNode node, Expression predicate)
                     || layout.getLayout().getColumns().get().containsAll(columnHandles);
         };
     }
+
+    private boolean planChanged(TableScanNode oldTableScan, TableScanNode rewrittenTableScan, Expression oldPredicate, Expression rewrittenPredicate)
+    {
+        if (!new ExpressionEquivalence(metadata, sqlParser).areExpressionsEquivalent(session, oldPredicate, rewrittenPredicate, symbolAllocator.getTypes())) {
```

Comment:
Do we need full `ExpressionEquivalence` here? Symbol names should be the same. Unless `resultingPredicate` is different each time `pickLayout` is called, then I think we can compare expressions directly.

Reply:
see https://github.com/Teradata/presto/pull/592 and in particular https://github.com/Teradata/presto/pull/592#discussion_r124269935

Reply:
Sure, it might change after one execution of `PickTableLayout` rule, especially since the predicate is rewritten by:
```
        Expression resultingPredicate = combineConjuncts(
                decomposedPredicate.getRemainingExpression(),
                DomainTranslator.toPredicate(layout.getUnenforcedConstraint().transform(assignments::get)));
```
, but the second call to the rule (when nothing changed) should produce exactly the same predicate (it should be deterministic). I am good either way.

Reply:
Leaving this as is.  It was introduced to fix an infinite loop where predicates were reordered with every iteration.